### PR TITLE
GH-268: Update libs to latest patch versions

### LINKED ISSUE #268: Upgrade libs for the 2021.1.2 release
We are out-of-date for a few dep. libs. Scan through and update all relevant lib versions.

- https://github.com/spring-cloud/stream-applications/blob/main/stream-applications-build/pom.xml
- https://github.com/spring-cloud/stream-applications/blob/main/applications/stream-applications-core/pom.xml

### PR DESCRIPTION
**Pull request by onobc**
Fixes #268

### CODE REVIEW

File: applications/source/file-source/README.adoc [hunk 1]
```diff
@@ -36,6 +36,43 @@ $$directory$$:: $$The directory to poll for new files.$$ *($$File$$, default: `$
 $$filename-pattern$$:: $$A simple ant pattern to match files.$$ *($$String$$, default: `$$<none>$$`)*
 $$filename-regex$$:: $$A regex pattern to match files.$$ *($$Pattern$$, default: `$$<none>$$`)*
 $$prevent-duplicates$$:: $$Set to true to include an AcceptOnceFileListFilter which prevents duplicates.$$ *($$Boolean$$, default: `$$true$$`)*
+
+=== metadata.store.dynamo-db
```
Comment (onobc) [inline applications/source/file-source/README.adoc:40]: This was updated as part of running the build.

File: applications/source/http-source/pom.xml [hunk 2]
```diff
@@ -12,6 +12,7 @@
         <groupId>org.springframework.cloud.stream.app</groupId>
         <artifactId>stream-applications-core</artifactId>
         <version>3.2.1-SNAPSHOT</version>
+	    <relativePath/>
```
Comment (onobc) [inline applications/source/http-source/pom.xml:15]: I will fix these formatting issues across the PR in another commit shortly.

File: applications/source/file-source/README.adoc (hunk 1 above)
Comment (artembilan) [reply applications/source/file-source/README.adoc:40]: I guess this is great and exactly what we would expect from `flie-source` in regards to its configuration properties.

And if this is really a result of just generation, then it really looks good with its dividing to paragraphs according nested properties classes 😄

File: applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceTests.java [hunk 3]
```diff
@@ -20,24 +20,37 @@
...
 import org.springframework.context.annotation.Import;
 import org.springframework.integration.test.mail.TestMailServer;
 import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-
 /**
+ * Lightweight integration test to verify using {@link MailSupplierConfiguration} in a [@code @SpringBootApplication}.
```
Comment (artembilan) [inline applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceTests.java:37]: Typo for the `{@code }` block.

File: applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020-2021 the original author or authors.
```
Comment (artembilan) [inline applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java:2]: This is already for not `2021`...

I'm not sure in legal matter, but general rule is to update the Copyright year to the current if you make some changes to it or this is just a new class for the project at all.

Other projects may have different rules. For example I heard that Spring Data team runs a script to update all the classes in the project to a new year.
In Spring Integration we go an approach to update via Gradle when there are some local changes to classes.

So, I'd prefer to have these updated, but I don't know rules for this project.

@sobychacko ?

File: applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.test.mail.TestMailServer;
+import org.springframework.messaging.Message;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MailSourceOldBrokenTests {
+
+	private static TestMailServer.MailServer MAIL_SERVER;
```
Comment (artembilan) [inline applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java:39]: This is removed already in Spring Integration `6.0` in favor of GreenMail library: https://github.com/greenmail-mail-test/greenmail.
I don't say to revise and fix this immediately, but need to be ready for consequences 😄

**Review by artembilan — CHANGES_REQUESTED** [review]
Just a couple general notes.
I will leave a decision to merge or not to @sobychacko 

Thanks

File: applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java (hunk 4 above)
Comment (onobc) [reply applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java:2]: Agreed, I will update and simply missed this. I update the copyrights anywhere I see them (normally).

File: applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java (hunk 5 above)
Comment (onobc) [reply applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java:39]: I will create a ticket in this repo to update the same based on this fact and mark it for the 2022 milestone.

Comment (onobc) [reply applications/source/mail-source/src/test/java/org/springframework/cloud/stream/app/source/file/MailSourceOldBrokenTests.java:39]: https://github.com/spring-cloud/stream-applications/issues/270

**Review by artembilan — APPROVED** [review]

**Review by sobychacko — APPROVED** [review]

**Pull request merged**